### ISSUE TITLE
Quest to Subquest alteration

### ISSUE THREAD

**Issue by Rantarian** [League-of-Foundry-Developers/foundryvtt-forien-quest-log#29]
There's no simple way of transforming a side-quest into a sub-quest of a larger quest. It would be very helpful to make it possible to do so through the interface.

**Comment by Shuggaloaf** [comment]
I would also love to see this functionality. 

After making a bunch of quests, it occurred to me I could have grouped some of them under 1 "main" quest. A way to change this after the fact would be a nice feature to have.

**Comment by Shuggaloaf** [comment]
@typhonrt Just finished watching the video you did on the rework. (Thank you for that btw) Player creation and Trusted Player edit brought me back to this issue. I know we spoke somewhat about this on Discord.

Just thought of a highly useful quest workflow (for some games) if this was an available feature. Trusted Player creates quest, adds objectives, etc. GM knows that this is actually a sub-quest of The Main Quest. GM can set this new quest as a sub-quest of The Main Quest. Players do not see this is a subquest since that's how you have it setup now, but GM does. This allows the GM to more easily track those subquests as part of the main and saves the GM a little effort in not having to create subquests ahead of time. 

Most importantly, it allows players the ability to create them as the campaign unfolds and gives them more agency and participation as well as letting them log the subquests and details _as they see it_. I find this to be invaluable for not only player involvement, but it also shows the players' level of understanding. The GM now knows if they are seeing events differently, missing stuff, etc.. I also personally find it easier to follow things and kind of reinforce events in my memory if I write it down myself. 

So a second plea with a little more, hopefully constructive, info on the value of this feature. :-)

**Comment by typhonrt** [comment]
@Shuggaloaf Yeah the way I'm thinking of implementing this feature is to provide a selection icon next to where the parent quest area is that is only visible for GM level users. Trusted players w/ edit won't see this icon and can't reassign quests / subquest connections. Click on the icon and a dialog pops up with an auto-complete area to type in the new potential parent quest name which then populates a drop down box. Select that new parent quest and FQL goes about reassigning the quest links. 

Do consider joining the [TyphonJS Discord](https://discord.gg/mnbgN8f) too which is a great place to brainstorm new ideas around all of this and other module extensions. That might provide a better spitballing forum then enhancement issues can be filed as appropriate, etc.

**Comment by Shuggaloaf** [comment]
> Do consider joining the [TyphonJS Discord](https://discord.gg/mnbgN8f) too which is a great place to brainstorm new ideas around all of this and other module extensions. That might provide a better spitballing forum then enhancement issues can be filed as appropriate, etc.

Thanks for the Discord invite! There are times when a spitballing-style discussion would definitely be more appropriate and I was actually thinking the same thing a few days ago. I had previously commented on this thread so just wanted to come back and mention that here since the issue already was opened.

To clarify for all of us - when you say:
> Click on the icon and a dialog pops up with an auto-complete area to type in the new potential parent quest name which then populates a drop down box
It sounds like this is basically saying - **Step 1:** Type Parent Quest name; **Step 2:** Parent Quest from Step 1 is now available in a dropdown and can be selected. 
Is that correct? If so, what if the Parent Quest already exists? 

Thanks @typhonrt

**Comment by typhonrt** [comment]
> It sounds like this is basically saying - **Step 1:** Type Parent Quest name; **Step 2:** Parent Quest from Step 1 is now available in a dropdown and can be selected.
> Is that correct? If so, what if the Parent Quest already exists?

The select box will populate with existing quests based on autocomplete. So, it's not creating a new quest, but assigning the parent quest by selecting a quest that already exists. The autocomplete angle to search for existing quests is pertinent especially when there are many quests to choose from. IE it could get unwieldy otherwise to list all potential quest targets for the assignment.

I'll likely be using [tail-select](https://getbutterfly.com/tail-select/) or similar more advanced select component.

**Comment by Shuggaloaf** [comment]
OK I got what you mean now - the dropdown box _itself_ allows autocomplete typing. I was thinking you meant 2 separate fields, a text input box and a dropdown. Sounds like a great solution! You always seem to go the extra little bit to make things more use-friendly. Definitely appreciate that!

**Comment by Jagusti** [comment]
Would this support promoting or altering subquests to quests? 
(I can see a future where all quests are just quests, and any quest can have an arbitrary link to a single other quest that is effectively its parent, which, when that is added effectively turns it into a subquest, but really its just a quest with a nominated linked quest.)

**Comment by typhonrt** [comment]
Hi @Jagusti! Indeed all quests are just a plain quest. They become a subquest in the GUI when they have a "parent quest". Indeed I'd like to organize things in the future where there is more that a one to many connection. Luckily that is just opening up the way the data is stored. 

With this initial quest to subquest alteration enhancement though there will be a way to reassign the parent of a quest. Remove the parent and the "subquest" becomes a new top level parent. Add a parent then that quest becomes a subquest.
